scripts/storage-emulator-integration/tests.ts

```diff
@@ -340,6 +340,146 @@ describe("Storage emulator", () => {
         });
       });
 
+      describe("#copy()", () => {
+        const copyDestinationFile = "copied_file";
+
+        it("should copy the file", async () => {
+          await testBucket.upload(smallFilePath);
+
+          const file = testBucket.file(copyDestinationFile);
+          const [, resp] = await testBucket.file(smallFilePath.split("/").slice(-1)[0]).copy(file);
+
+          expect(resp)
+            .to.have.all.keys(["kind", "totalBytesRewritten", "objectSize", "done", "resource"])
+            .and.include({
+              kind: "storage#rewriteResponse",
+              totalBytesRewritten: String(SMALL_FILE_SIZE),
+              objectSize: String(SMALL_FILE_SIZE),
+              done: true,
+            });
```

Comment:
Writing this has made me think a bug might exist in the types for `@google-cloud/storage`...?

The return type from [`#copy()`](https://github.com/googleapis/nodejs-storage/blob/0c1fa3934a52a608366a8c6c798c43516dd03dbf/src/file.ts#L836-L845) is [`[File, Metadata]`](https://github.com/googleapis/nodejs-storage/blob/0c1fa3934a52a608366a8c6c798c43516dd03dbf/src/file.ts#L318) but from looking at the [source](https://github.com/googleapis/nodejs-storage/blob/0c1fa3934a52a608366a8c6c798c43516dd03dbf/src/file.ts#L1080-L1103) and the [docs](https://googleapis.dev/nodejs/storage/latest/global.html#CopyResponse) it should be more like `[File, { resource: Metadata }]`.

Though I know the docs/types for `@google-cloud/storage` are a bit hit and miss... 🙃

Reply:
It's very possible there is a bug in that SDK and/or the docs, we don't have any impact on that codebase, so I'd recommend filing a bug under that repo